GiEnJul.Test/ControllerTests/AdminControllerTest.cs

```diff
@@ -0,0 +1,115 @@
+﻿using GiEnJul.Clients;
+using GiEnJul.Controllers;
+using GiEnJul.Infrastructure;
+using GiEnJul.Models;
+using GiEnJul.Repositories;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GiEnJul.Test.ControllerTests
+{
+    public class AdminControllerTest : BaseControllerTest
+    {
+        private Mock<IEventRepository> mockEventRepo { get; set; }
+        public Mock<IGiverRepository> mockGiverRepo { get; private set; }
+        public Mock<IRecipientRepository> mockRecipientRepo { get; private set; }
+        public Mock<IPersonRepository> mockPersonRepo { get; private set; }
+        public Mock<IConnectionRepository> mockConnectionRepo { get; private set; }
+
+        private readonly Mock<IEmailClient> emailClientMock;
+
+        public Mock<IConfiguration> configMock { get; private set; }
+        public readonly Settings settings;
+
+        private AdminController _controller;
+
+        public AdminControllerTest()
+        {
+            mockEventRepo = new Mock<IEventRepository>();
+            mockGiverRepo = new Mock<IGiverRepository>();
+            mockRecipientRepo = new Mock<IRecipientRepository>();
+            mockPersonRepo = new Mock<IPersonRepository>();
+            mockConnectionRepo = new Mock<IConnectionRepository>();
+            emailClientMock = new Mock<IEmailClient>();
+            configMock = new Mock<IConfiguration>();
+            settings = new Settings(configMock.Object);
+            _controller = new AdminController(mockEventRepo.Object,
+                                              mockGiverRepo.Object,
+                                              mockRecipientRepo.Object,
+                                              mockPersonRepo.Object,
+                                              mockConnectionRepo.Object,
+                                              _log,
+                                              _mapper,
+                                              emailClientMock.Object,
+                                              settings);
+        }
+
+        public void Dispose()
+        {
+            mockEventRepo.VerifyNoOtherCalls();
```

Comment:
Could add `mockGiverRepo.VerifyNoOtherCalls()` This will throw an exception if any calls to the GiverRepo is done, other than those expected. Expected being defined with `mockGiverRepo.Verify(x => x.MyMethod...`